src/main/java/reactor/netty/resources/DefaultLoopEpoll.java

```diff
@@ -50,6 +54,12 @@
 		if (channelClass.equals(DatagramChannel.class)) {
 			return (CHANNEL) new EpollDatagramChannel();
 		}
+		if (channelClass.equals(DomainSocketChannel.class)) {
+			return (CHANNEL) new EpollDomainSocketChannel();
+		}
+		if (channelClass.equals(ServerDomainSocketChannel.class)) {
+			return (CHANNEL) new EpollServerDomainSocketChannel();
```

Comment:
I think that is orthogonal to this PR, but I'm wondering about the throwing in both `getChannel` and `channelType`. AFAICS when producing a `ChannelFactory` in `xxxConfig#connectionFactory`, the lambda representing the factory calls `channelType`, which can throw. The result is then wrapped by a call to `getChannel`, which can also throw. But that exception comes late in the cycle, _after_ having called the configuration methods, right ? Or is the exception surfacing right away when building the client / binding the server?

I was wondering if a null-object pattern could be used instead to signal an illegal combination of channel type and transport earlier. The `channelType` would produce a special class and the factory  could check it eagerly, in order to throw from the configuration phase?

of course this is all moot if the factory MUST be lazy (ie, the channel type can be changed BETWEEN a call to `xxxConfig#connectionFactory` and the final build/bind step. This is also less of an issue if the throwing `ChannelFactory` can never make it past the build/bind step, ie the resulting client or server could never start.

Reply:
By default `LoopResources#onChannel` method invokes the default implementation used by Reactor Netty.
It throws exception to signal that the default implementation does not know what to do with other class types. Of course the user can provide his/her own implementation that will be able to handle other class types.